mail/libraries/lite_to_edifact_converter.py

```
@@ -96,34 +111,40 @@ def licences_to_edifact(licences: QuerySet, run_number: int) -> str:
+                for g, commodity in enumerate(licence_payload.get("goods"), start=1):
+                    line_no += 1
                     GoodIdMapping.objects.get_or_create(
                         lite_id=commodity["id"], licence_reference=licence.reference, line_number=g
                     )
-                    edifact_file += "\n{}\\line\\{}\\\\\\\\\\{}\\Q\\{}\\{}".format(
-                        i,
+                    controlled_by = "Q"  # usage is controlled by quantity only
+                    edifact_file += "\n{}\\line\\{}\\\\\\\\\\{}\\{}\\\\{:03d}\\\\{}".format(
+                        line_no,
                         g,
-                        commodity.get("description"),
+                        commodity.get("name"),
+                        controlled_by,
                         UnitMapping.convert(commodity.get("unit")),
                         int(commodity.get("quantity")) if commodity.get("unit") == "NAR" else commodity.get("quantity"),
                     )
             if licence_payload.get("type") in LicenceTypeEnum.OPEN_LICENCES:
-                i += 1
+                line_no += 1
                 edifact_file += "\n{}\\line\\1\\\\\\\\\\Open Licence goods - see actual licence for information\\".format(
-                    i
+                    line_no
                 )
-        i += 1
-        edifact_file += "\n{}\\end\\licence\\{}".format(i, i - start_line)
-    i += 1
+        line_no += 1
+        edifact_file += "\n{}\\end\\licence\\{}".format(line_no, line_no - start_line)
+    line_no += 1
     edifact_file += "\n{}\\fileTrailer\\{}".format(
-        i, licences.count() + licences.filter(action=LicenceActionEnum.UPDATE).count()
+        line_no, licences.count() + licences.filter(action=LicenceActionEnum.UPDATE).count()
     )
+
+    errors = validate_edifact_file(edifact_file)
+    if errors:
+        logging.error(f"File content not as per specification, {errors}")
+        return None
```

Might be better to raise an Exception here instead of logging manually and returning with `None`.

I think (although I am not sure) that this would take care of the -

```
if file_content is None:
         return None
```

Earlier in this PR.

---

:+1: